validation/src/lib.rs

```diff
@@ -95,13 +97,18 @@ pub trait TableRouter: Clone {
 	type Error: std::fmt::Debug;
 	/// Future that resolves when candidate data is fetched.
 	type FetchValidationProof: IntoFuture<Item=PoVBlock,Error=Self::Error>;
+	/// Future that resolves when a block erasure chunk is fetched.
+	type FetchBlockChunk: IntoFuture<Item=ErasureChunk, Error=Self::Error>;
 
 	/// Call with local candidate data. This will make the data available on the network,
 	/// and sign, import, and broadcast a statement about the candidate.
-	fn local_collation(&self, collation: Collation, outgoing: OutgoingMessages);
+	fn local_collation(&self, collation: Collation, outgoing: OutgoingMessages, chunks: (ValidatorIndex, &Vec<ErasureChunk>));
 
 	/// Fetch validation proof for a specific candidate.
 	fn fetch_pov_block(&self, candidate: &CandidateReceipt) -> Self::FetchValidationProof;
+
+	/// Fetch the erasure-encoded chunk for a specific candidate block.
+	fn fetch_erasure_chunk(&self, relay_parent: Hash, candidate_hash: Hash, chunk_id: u64) -> Self::FetchBlockChunk;
```

Comment:
this is defined and seems to cause a lot of turmoil in the networking code - but is it actually used anywhere?

1. it seems at-odds with the gossip protocol. we should have a direction and stick with it.
2. the networking code underlying the implementation here is messy. I don't like the invasion into `dispatch_pending_requests` and the extra `in_flight_chunk_requests`.

If it's not used, then I'd suggest removing it for this PR.